### DISCUSSION TITLE
Connecting to BMD Arduino Shield

### DISCUSSION THREAD

**Discussion by just-me-jb** [FanatiQS/wireless_atem_camera_control#3] [Q&A]
Hi,
really cool project! Thanks for that.
Could run it without Arduino shield connected and works es expected. But no way to communicate with the shield, Wemos D1 mini connected with a level shifter to the shield, no I2C communication at all.
Could you give a hint what might be the problem?

Cheers

Jens

**Comment by FanatiQS** [comment]
Hi Jens,
Glad you like the project :)

First thing to ask, have you enabled SDI communication in "user_config.h"? It has 2 commented out definitions for the pins to use for I2C. Note that the default pin definitions in "user_config.h" is reversed compared to what the ESP8266 documents. This is just because the way I have mounted it to the SDI shield and it is fine to use any pin since I2C on the ESP8266 is all done in software (bitbanging style) and is therefore not limited to specific pins.

If it is enabled in "user_config.h" and the BMDSDIControl library is installed (with slight modification to compile) but without the shield connected, the ESP should get a WDT (watchdog timer) after only a few seconds of trying to connect to the SDI shield. If you get that WDT, that means you at least have everything working in software.

At that point I would create a separate sketch that blinks the I2C pins on and off every few seconds and measure the voltage between the pins and ground both before and after the logic level shifter. Before the shifter, it should be 3.3v and after there should be 5v.

Some extra notes that that might help:
* Since the Wemos D1 mini is powered by 5v and the SDI shield by 7-12v, they either have to be powered by separate sources or use some voltage regulator / buck converter.
* The ESP and logic level shifter should share ground with the SDI shield. I had issues with communication before I learned that.

That is all I can think of at the moment. If it doesn't help, let me know and I'll have to think a little harder xD
Also, let me know if you get it working too :)

Good luck!
/Gustav
